core/tests/protractor_utils/TopicsAndSkillsDashboardPage.js

```diff
@@ -205,17 +228,53 @@ var TopicsAndSkillsDashboardPage = function() {
     return await waitFor.pageToFullyLoad();
   };
 
+  this.filterTopicsByKeyword = async function(keyword) {
+    await waitFor.visibilityOf(
+      topicFilterKeywordField,
+      'Topic Dashboard keyword filter parent taking too long to appear.');
+    var filterKeywordInput = topicFilterKeywordField.element(
+      by.css('.select2-search__field'));
+    await waitFor.visibilityOf(
+      filterKeywordInput,
+      'Topic Dashboard keyword filter taking too long to appear.');
+
+    await filterKeywordInput.click();
+    await filterKeywordInput.sendKeys(keyword);
+    await filterKeywordInput.sendKeys(protractor.Key.RETURN);
+  };
+
+  this.filterTopicsByCategory = async function(keyword) {
+    await waitFor.visibilityOf(
+      topicFilterCategoryField,
+      'Topic Dashboard category taking too long to appear.');
+
+    await topicFilterCategoryField.click();
+    await browser.driver.switchTo().activeElement().sendKeys(keyword + '\n');
+  };
+
+  this.resetTopicFilters = async function() {
+    await waitFor.visibilityOf(
+      topicResetFilters, 'Reset button taking too long to be clickable');
+    await topicResetFilters.click();
+  };
+
   this.deleteTopicWithIndex = async function(index) {
-    var elems = await deleteTopicButtons;
+    await waitFor.visibilityOf(topicsTable,
+      'Topic table taking too long to appear.');
+    var topics = await element.all(by.css('.protractor-test-topic-edit-box'));
+    var deleteTopicButton = element(
+      by.css('.protractor-test-delete-topic-button'));
+    var topicEditOptionBox = topics[index];
+    await browser.actions().mouseMove(topicEditOptionBox).perform();
     await waitFor.elementToBeClickable(
-      elems[0],
+      deleteTopicButton,
       'Delete Topic button takes too long to be clickable');
-    await elems[0].click();
-
+    await deleteTopicButton.click();
     await waitFor.elementToBeClickable(
       confirmTopicDeletionButton,
       'Confirm Delete Topic button takes too long to be clickable');
     await confirmTopicDeletionButton.click();
+    await this.get();
     await waitFor.pageToFullyLoad();
```

Comment:
Does `this.get()` already wait for the page to load?

Reply:
Done.